ojdbc-provider-jackson-oson/src/main/java/oracle/jdbc/provider/oson/OsonParser.java

```diff
@@ -0,0 +1,703 @@
+        return JsonToken.VALUE_FALSE;
+
+      case VALUE_TIMESTAMP:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_TIMESTAMP");
+        return JsonToken.VALUE_STRING;
+
+      case VALUE_TIMESTAMPTZ:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_TIMESTAMPTZ");
+        return JsonToken.VALUE_STRING;
+        
+      case VALUE_BINARY:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_BINARY");
+        return JsonToken.VALUE_STRING;
+        
+      case VALUE_INTERVALDS:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_INTERVALDS");
+        return JsonToken.VALUE_STRING;
+        
+      case VALUE_INTERVALYM:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_INTERVALYM");
+        return JsonToken.VALUE_STRING;
+
+      default:
+        throw new IllegalStateException("event: " + currentEvent);
+    }
+
+  }
+
+  /**
+   * Get the current token.
+   * @return
+   */
+  @Override
+  public JsonToken currentToken() {
+    return getCurrentToken();
+  }
+
+  /**
+   * Get the current Token ID.
+   */
+  @Override
+  public int getCurrentTokenId() {
+    if(DEBUG) logger.log(Level.FINEST, "getCurrentTokenId");
+    JsonToken jt;
+    if ((jt = OSON_EVENT_TO_JSON_TOKEN.get(currentEvent)) != null) {
+      return jt.id();
+    } else if( currentEvent == OracleJsonParser.Event.VALUE_DECIMAL ) {
+      return parser.isIntegralNumber() ? JsonToken.VALUE_NUMBER_INT.id() : JsonToken.VALUE_NUMBER_FLOAT.id();
+    }
+
+    return -1;
+  }
+
+  /**
+   * Get the current Token ID.
+   */
+  @Override
+  public int currentTokenId() {
+    if(DEBUG) logger.log(Level.FINEST, "getCurrentTokenId");
+    JsonToken jt;
+    if ((jt = OSON_EVENT_TO_JSON_TOKEN.get(currentEvent)) != null) {
+      return jt.id();
+    } else if( currentEvent == OracleJsonParser.Event.VALUE_DECIMAL ) {
+      return parser.isIntegralNumber() ? JsonToken.VALUE_NUMBER_INT.id() : JsonToken.VALUE_NUMBER_FLOAT.id();
+    }
+
+    return JsonTokenId.ID_NO_TOKEN;
+  }
+
+  /**
+   * Get the current Token ID.
+   */
+  @Override
+  public boolean hasCurrentToken() {
+    if(DEBUG) logger.log(Level.FINEST, "hasCurrentToken");
+    return currentEvent != null;
+  }
+
+  /**
+   * Checks if the provided id matches the ID FIELD.
+   */
+  @Override
+  public boolean hasTokenId(final int id) {
+    if(DEBUG) logger.log(Level.FINEST, "hasTokenId( " + id + " )");
+    if (id == JsonTokenId.ID_FIELD_NAME) {
+      return currentEvent == OracleJsonParser.Event.KEY_NAME;
+    }
+    if(id == JsonTokenId.ID_STRING) {
+      return currentEvent == OracleJsonParser.Event.VALUE_STRING;
+    }
+
+    return false;
+  }
+
+  /**
+   * Checks if the current token matches the provided JsonToken.
+   *
+   * @param jsonToken The JsonToken to check against.
+   * @return true if the current token matches the provided JsonToken, false otherwise.
+   */
+  @Override
+  public boolean hasToken(JsonToken jsonToken) {
+    if(DEBUG) logger.log(Level.FINEST, "hasToken( " + jsonToken + " )");
```

Comment:
The methods hasToken and hasTokenId have the same description, but they don't seem to do the same.